include/irr/asset/IAsset.h

```diff
@@ -173,6 +173,17 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
+		//! restore() is not recursive!
+		virtual bool restore(IAsset* _other)
+		{
+			if (!this->isADummyObjectForCache())
+				return false;
+			assert(!_other->isADummyObjectForCache());
```

Comment:
would be useful if we called it `swapContents` and had the two assets swap stuff around